engine/schema/src/main/java/com/cloud/vm/dao/VMInstanceDaoImpl.java

```diff
@@ -613,7 +613,7 @@ boolean ifStateUnchanged(State oldState, State newState, Long oldHostId, Long ne
         if (backupOfferingId != null) {
             sc.setParameters("backup_offering_id", backupOfferingId);
         }
-        return listBy(sc);
+        return listIncludingRemovedBy(sc);
```

Comment:
@Pearl1594 `listByZoneWithBackups` is also used by `BackupSyncTask`. Can there be a case of usage event being published for a deleted VM or is it fine to publish events with type`EVENT_VM_BACKUP_USAGE_METRIC` for deleted VMs?

Reply:
@shwstppr It is expected for usage events to be published for VMs that are destroyed as long as there are backups available for that VM - which is checked by the backupsize - i.e., only if backup size > 0 will `EVENT_VM_BACKUP_USAGE_METRIC` event be published .

Reply:
The delete backup offering API will fail @Pearl1594

Reply:
@rhtyd deletion of the offering will fail if the they aren't assigned to a VM, but are associated to a backup.